src/large_pages/node_large_page.cc

```diff
@@ -44,38 +78,25 @@
 #include <sys/mman.h>
 #if defined(__FreeBSD__)
 #include <sys/sysctl.h>
-#include <sys/user.h>
 #elif defined(__APPLE__)
 #include <mach/vm_map.h>
 #endif
-#include <unistd.h>  // getpid
 
 #include <climits>  // PATH_MAX
-#include <clocale>
-#include <csignal>
 #include <cstdlib>
 #include <cstdint>
 #include <cstring>
 #include <string>
 #include <fstream>
-#include <iostream>
-#include <vector>
-
-// The functions in this file map the text segment of node into 2M pages.
-// The algorithm is simple
-// Find the text region of node binary in memory
-// 1: Examine the /proc/self/maps to determine the currently mapped text
-// region and obtain the start and end
-// Modify the start to point to the very beginning of node text segment
-// (from variable nodetext setup in ld.script)
-// Align the address of start and end to Large Page Boundaries
-//
-// 2: Move the text region to large pages
-// Map a new area and copy the original code there
-// Use mmap using the start address with MAP_FIXED so we get exactly the
-// same virtual address
-// Use madvise with MADV_HUGEPAGE to use Anonymous 2M Pages
-// If successful copy the code there and unmap the original region.
+
+#if !defined(__APPLE__)
+#define LPSTUB_SECTION __attribute__((__section__("lpstub")))
+#else
+#define LPSTUB_SECTION __attribute__((__section__("__TEXT,__lpstub")))
+#endif
+#define IN_LPSTUB               \
+  LPSTUB_SECTION                \
+  __attribute__((__noinline__))
```

Comment:
`always_inline` would do the job too for the `MemoryMapPointer` class, right? I think that makes more sense given the nature of the class

Reply:
Done.